cloudify_ansible_sdk/sources.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class AnsibleSource(object):
+
+    def __init__(self, source_dict):
+
+        source_dict = source_dict or {}
+        self.groups = {}
+
+        for group_name, group_config in source_dict.items():
+            self.insert_group(group_name, group_config)
+
+    def insert_group(self, name, config=None):
+        config = config or {}
+        self.groups[name] = AnsibleHostGroup(name, **config)
+
+    def merge_source(self, ansible_source):
+        for group_name, group in ansible_source.groups.items():
```

Comment:
@EarthmanT Do not you think using `iteritems` is more efficient here than items()  ?

Reply:
spoke about it offline :)

Reply:
Thanks